xapian-letor/ranker/listnet_ranker.cc

```diff
@@ -120,12 +120,13 @@ update_parameters(vector<double> &new_parameters, const vector<double> &gradient
 }
 
 void
-ListNETRanker::train(const std::vector<Xapian::FeatureVector> & training_data) {
+ListNETRanker::train(const std::vector<vector<Xapian::FeatureVector>>&
+		     training_data) {
```

Comment:
The opening `{` for an out-of-line function definition should really be on a fresh line by Xapian coding conventions (this is currently often wrong in letor code).

`using namespace std;` is already active in this file, so you don't need the `std::` qualification - if you remove that and fix the `{` location then there's (just) no need to wrap this.

Reply:
fixed in a fixup commit.